training_platform/server/service.py

```diff
@@ -47,25 +49,33 @@ def receiveMessage(self, msg, sender):
             self.send(sender, response)
 
         elif isinstance(msg, PlayerClientsMsg):
+            self.log(f"Received PlayerClientsMsg message with the following players_clients: {msg.players_clients}")
             self.players_clients = msg.players_clients
             self.ready_to_start = True
 
         elif isinstance(msg, StartEnvMsg):
+            self.log(f"Received StartEnvMsg")
             if not self.ready_to_start:
                 self.send(sender, EnvNotReadyToStartMsg())
+                self.log(f"Sent EnvNotReadyToStartMsg to sender")
                 return
             self.ready_to_start = False
             self.who_started_game = sender
+            self.log(f"New who_started_game={self.who_started_game}")
+            self.notify_on_end = msg.notify_on_end
+            self.log(f"New self.notify_on_end={self.notify_on_end}")
             for player in self.players_clients.keys():
                 self.before_first_move[player] = True
             self.environment.reset()
-            self.log(f"Launched game with following players and clients: {self.players_clients}")
+            self.log(f"Started game with following players and clients: {self.players_clients}")
             current_client = self.players_clients[self.environment.current_player]
             self.send(current_client, YourTurnMsg(self.environment.current_board, self.environment.allowed_actions))
+            self.log(f"Sent YourTurnMsg to current_client: {current_client}")
             self.send(self.who_started_game, EnvStartedMsg())
-
+            self.log(f"Sent EnvStartedMsg to self.who_started_game: {self.who_started_game}")
 
         elif isinstance(msg, TakeActionMsg):
+            self.log(f"Received TakeActionMsg")
             self.before_first_move[self.environment.current_player] = False
             self.environment.make_move(msg.action)  # It implicitly makes next player current player
             # State update for GUI clients TODO: send StateUpdateMsg only to GUI clients
```

Comment:
to juz jest zrobione z tego co widze wiec komentarz do wyrzucenia

Reply:
nie, na razie wysyła state update do wszystkich klientów, w tym RL, którzy po prostu to ignorują, chcę zrobić żeby StateUpdate był wysyłany tylko do klientów typu gui.

Reply:
Nie jest zrobione, póki co game manager wysyła StateUpdateMsg do wszystkich klientów zamiast tylko do GUI klientów